### PR TITLE
Adds `react-hook-form` and `zod` for improved validation

### LINKED ISSUE #38: Introduce `react-hook-form`

### PR DESCRIPTION
**Pull request by douglance**
closes #38 

The goal of this PR is to make it harder for end-users to misconfigure their Orbit chains. To accomplish that, this PR adds `zod` for validating data with `react-hook-form` for integrating `zod` into the form validation logic.

## changes

- [x] adds react-hook-form 
- [x] adds zod validation for forms

## how to test

try adding bad inputs to the fields (empty fields, strings into number fields, non-addresses to address inputs, etc.)

### CODE REVIEW

**Review by dewanshparashar — COMMENTED** [review]
Good choice with `react-hook-form` and `zod`. 
Need some clarifications + minor suggestions.

File: src/types/RollupContracts.ts [hunk 1]
```diff
@@ -10,12 +10,7 @@ export type RollupContracts = {
   deployedAtBlockNumber: number;
 };
 
-export type Validator = {
-  address: string;
-  privateKey?: string;
-};
-
-export type BatchPoster = {
+export type ConfigWallet = {
```
[the remark this answers is not in the document]
Comment (dewanshparashar) [reply src/types/RollupContracts.ts:13]: Also renaming `validators` -> `wallets` within the code, any reason for that? To me they just look like simple `addresses`.

File: src/components/TextInputWithInfoLink.tsx [hunk 2]
```diff
@@ -1,39 +1,47 @@
-import { FC, InputHTMLAttributes } from 'react';
+import { UseFormRegisterReturn, InternalFieldName } from 'react-hook-form';
 import { InfoCircleWithTooltip } from './InfoCircleWithTooltip';
+import { type } from 'os';
```
Comment (dewanshparashar) [inline src/components/TextInputWithInfoLink.tsx:3]: Unused import

File: package.json [hunk 3]
```diff
@@ -23,13 +24,15 @@
...
     "react-dom": "18.2.0",
+    "react-hook-form": "^7.45.4",
     "react-syntax-highlighter": "^15.5.0",
     "react-transition-group": "^4.4.5",
     "use-clipboard-copy": "^0.2.0",
     "use-query-params": "^2.2.1",
     "viem": "^1.6.0",
     "wagmi": "^1.3.9",
-    "web3modal": "^1.9.12"
+    "web3modal": "^1.9.12",
```
Comment (chrstph-dvx) [inline package.json:34]: I know it's only an update here, but any reason we use `web3modal` over rainbowkit?

File: src/components/SetValidators.tsx [hunk 4]
```diff
@@ -1,46 +1,83 @@
...
   useEffect(() => {
-    setValidators((prev) => {
-      if (prev.length < validatorCount) {
-        return [...prev, ...Array.from({ length: validatorCount - prev.length }, getRandomWallet)];
-      } else {
-        return prev.slice(0, validatorCount);
-      }
-    });
-  }, [validatorCount]);
+    let newWallets = wallets;
```
Comment (chrstph-dvx) [inline src/components/SetValidators.tsx:43]: Any reason we initialize to `wallets`? If I understand correctly, we overwrite it on the next few lines

**Review by chrstph-dvx — COMMENTED** [review]
Just did a first pass, I need to look into it more tomorrow.

**Review by chrstph-dvx — APPROVED** [review]

**Review by dewanshparashar — APPROVED** [review]

**Pull request merged**